### PR TITLE
Hoist non-static properties on withTheme HOC

### LINKED ISSUE #596: Make withTheme hoist static properties
For reference: https://github.com/react-community/react-navigation/issues/381

### PR DESCRIPTION
**Pull request by philpl**
Fix #596 

See for reference:

- https://github.com/reactjs/react-redux/issues/53
- https://github.com/reactjs/react-redux/issues/276

### CODE REVIEW

File: src/hoc/withTheme.js [hunk 1]
```diff
@@ -3,34 +3,49 @@
...
+const wrapWithTheme = (Component: ReactClass<any>) => {
+  const componentName = (
+    Component.displayName ||
+    Component.name ||
+    'Component'
+  )
 
-  state: { theme?: ?Object } = {};
-  unsubscribe: () => void;
+  class WithTheme extends React.Component {
```
Comment (geelen) [inline src/hoc/withTheme.js]: Is it possible to do `class WithTheme extends Component`? Would that mean statics are available by simple prototype inheritance?

Comment (philpl) [inline src/hoc/withTheme.js]: That's not a good idea, since we'd be overriding some original logic. I'd stick to the HOC-pattern :wink:

**Comment by philpl** [comment]
@brunolemos I understand you're upset and I apologise for stepping on your toes. Please do understand however that I've thought about this and opened a fresh PR since it's going to be merged against v2 and now intersects/conflicts with #710. I'm currently trying to move quickly towards the v2 release, and I thought it's safe to move your changes over to a new branch.

I'll add an attribution with your name to the changelog, and I hope this doesn't scare you off from contributing to SC in the future. Sorry again! 😅

**Comment by brunolemos** [comment]
@philpl I was just finishing to update this PR with the original commit plus yours, but nevermind, you merged it.

**Comment by philpl** [comment]
@brunolemos Hey, man. I'm sorry I went ahead, but I can't read your mind and guess what you're up to. Yes, this is unfortunate, but at the end of the day it's just code, so please don't be passive aggressive about this, because that doesn't help either of us, since it's not constructive.

I'm happy to buy you a beer and apologise in person, if we end up in the same city one day. We don't have to escalate this further and turn this into something bigger than it is. Sorry again, mate

**Comment by brunolemos** [comment]
@philpl It's constructive in the way that my intention is that you don't do that again with other  contributor or a project member like me. Besides that, no problem, keep moving fast.

**Pull request merged**